projects/packages/scheduled-updates/tests/php/class-wpcom-rest-api-v2-endpoint-update-schedules-test.php

```diff
@@ -940,6 +909,147 @@ public function test_add_log_unauthorized() {
 		$this->assertSame( 401, $result->get_status() );
 	}
 
+	/**
+	 * Test create item with paths.
+	 *
+	 * @covers ::create_item
+	 */
+	public function test_create_item_with_no_paths() {
+		$plugins  = array( 'gutenberg/gutenberg.php' );
+		$request  = new WP_REST_Request( 'POST', '/wpcom/v2/update-schedules' );
+		$schedule = $this->get_schedule();
+
+		// No health check paths.
+		unset( $schedule['health_check_paths'] );
+
+		$request->set_body_params(
+			array(
+				'plugins'  => $plugins,
+				'schedule' => $schedule,
+			)
+		);
+		$schedule_id = Scheduled_Updates::generate_schedule_id( $plugins );
+
+		// Successful request.
+		wp_set_current_user( $this->admin_id );
+		$result = rest_do_request( $request );
+
+		$this->assertSame( 200, $result->get_status() );
+		$this->assertSame( $schedule_id, $result->get_data() );
+
+		$option_paths = Scheduled_Updates::get_health_check_paths( $schedule_id );
+		$this->assertSame( array(), $option_paths );
+	}
+
+	/**
+	 * Test create item with paths.
+	 *
+	 * @covers ::create_item
+	 */
+	public function test_create_item_with_various_paths() {
+		$plugins = array( 'gutenberg/gutenberg.php' );
+		$paths   = array(
+			"a\nb",
+			'=',
+			'=',
+			' ',
+			"\ntest\t",
+			'*',
+			'ünicode',
+		);
+		$request = new WP_REST_Request( 'POST', '/wpcom/v2/update-schedules' );
+		$request->set_body_params(
+			array(
+				'plugins'  => $plugins,
+				'schedule' => $this->get_schedule( 'next Monday 8:00', 'weekly', $paths ),
+			)
+		);
+		$schedule_id = Scheduled_Updates::generate_schedule_id( $plugins );
+
+		// Successful request.
+		wp_set_current_user( $this->admin_id );
+		$result = rest_do_request( $request );
+
+		$this->assertSame( 200, $result->get_status() );
+		$this->assertSame( $schedule_id, $result->get_data() );
+
+		$option_paths = Scheduled_Updates::get_health_check_paths( $schedule_id );
+		$this->assertSame(
+			array(
+				'a_b',
+				'=',
+				'',
+				'test',
+				'*',
+				'ünicode',
```

Comment:
Should we allow path like only `=`, `*` or `' '` to pass the validation?

Reply:
Now, all the strings filtered away by `esc_url_raw` are not accepted.